### ISSUE TITLE
Slow performance running VM for first time

### ISSUE THREAD

**Issue by noahsmartin** [cirruslabs/tart#903]
Hi! Thanks for making this, it’s been a really useful tool for us. We have a few machines all on macOS 14.6.1 running tart 2.18.1 and for just one machine we see significantly longer time to boot a VM. We are running:

```
tart clone SRC DEST
tart run DEST
```

And timing how long until the VM is booted. For all but one machine it's around 15 seconds, but for one machine it's > 45 seconds. The hardware configuration of these machines are the same. I'm wondering if there are any debugging tools to use in tart (maybe a verbose mode?) that could tell us what step it’s getting stuck on? From activity monitor I can see the CPU isn't very busy while it's seemingly stuck. On other machines CPU usage increases the moment I launch `tart run`, but on this machine it takes a while before it gets going. The tart window is open during this time but is just black (not even the apple boot screen).

After the VM is launched one time it’s much faster to launch a second time, but for our workflow we care about this first launch time. Any advice on how to investigate or speed it up is appreciated!

**Comment by fkorotkov** [comment]
Hey @noahsmartin, is your `SRC` a remote VM which is pre-pulled? Do you also run multiple VMs on the same host?

Also how do you time the booting time? Just `tart ip DEST`?

**Comment by noahsmartin** [comment]
Thanks @fkorotkov 
SRC is a VM that I originally pulled with `tart clone URL SRC`, only one VM is running at a time on the host and it's usually a clone of the same SRC. We do use `tart ip DEST` to determine when it's booted, but it's such a large difference it's also clear it's taking longer just from watching how long the tart window is black before the apple boot screen

**Comment by fkorotkov** [comment]
Interesting. I think we can try to look for some system logs both on host and guest sides.

On the host you can open `Console.app`, start streaming and then do the VM boot. After stopping you can filter for Tart logs via `process:tart` query and for all Tart related logs. `Cmd+A` to select them all and `Cmd+C/Cmd+V` to share. We might see if there are particular gaps/hickups in the logs timeline.

On the guest side you can also open `Console.app` and see what `system.log` has for `BOOT_TIME` and if it's different from the other machines.

Finally, most of the issues we've seen were related to host DHCP handling. You might try to `sudo rm -f /var/db/dhcpd_leases` on the host and see if that improves the booting.

**Comment by noahsmartin** [comment]
@fkorotkov I added some time logs into tart and compiled it/ran with that version and saw that it was taking very little time to get to this line: https://github.com/cirruslabs/tart/blob/main/Sources/tart/Commands/Run.swift#L350

So it seemed like the issue was all after tart handed off to Virtualization.framework

I tried deleting the OCI image in tart and then pulled it again with `tart clone URL SRC_2` then when I tried to clone/run `SRC_2` it was back to normal speed! I'm not sure what went wrong the first time we cloned it that made it slow to boot any copy of it, but it is working fine now so I'll close the ticket. Hopefully this helps if anyone else encounters a similar issue!
